lib/views/sign_in.dart

```diff
@@ -31,6 +32,9 @@ class _SignInPageState extends State<SignInPage> {
       var res = await AuthService.login(getFormData());
       final parsed = LoginRes.fromJson(jsonDecode(res));
       if (parsed.success) {
+        SharedPreferences prefs = await SharedPreferences.getInstance();
```

Comment:
I'd move saving shared prefs to another method.
Couldn't saving token on login be asynchronous not to block UI?

Reply:
Does Dart have any event loop? I'm worried that asynchronous calls are discarded if they're are not waited for (like in Python). If Dart works anything like Javascript then it'd be fine. I'll leave it synchronous until there's a confirmation it'll work